### PR TITLE
feat: adding the ability to pull fiat rates

### PR DESCRIPTION
**Pull request by devinpearson**
This is the start of the rates module. 

currently it is able to pull rates from two sources. 

This will be expanded to get crypto rates from multiple sources and display them in any fiat currency

### CODE REVIEW

**Comment by karlbuys** [comment]
Just a personal take, I would rename classes like "ExchangeRateApi" to just "RateApi". Alternatively, FiatRateApi, CryptoRateApi

**Review by karlbuys — APPROVED** [review]

File: rates/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    // terms_of_use: String,
+    time_last_update_unix: u64,
+    #[serde(rename = "base_code")]
+    base: String,
+    #[serde(rename = "conversion_rates")]
+    rates: HashMap<String, f32>,
+}
+
+/// The ExchangeRateApi provider
+pub struct ExchangeRateApi {
```
[the remark this answers is not in the document]
Comment (devinpearson) [reply rates/src/lib.rs:46]: its the name of the service. exchangerate-api.com. the other one ive connected to is exchangerate.host. was just keeping the name shorter

File: rates/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+readme = "README.md"
+categories = ["cryptography::cryptocurrencies"]
+
+[lib]
+name = "walletd_rates"
+path = "src/lib.rs"
+
+[dependencies]
+thiserror = "1.0.40"
+"serde" = { version = "1.0.160", features = ["derive"] }
```
[the remark this answers is not in the document]
Comment (devinpearson) [reply rates/Cargo.toml:19]: seemed to just be the way i copied it across. ill remove them

**Comment by devinpearson** [comment]
So I should cut the name of the site out?  ExchangeRateApi is based on the site name. So is ExchangeRateHost. Those are the services names.

**Pull request merged**